src/components/Error/ErrorBoundary.tsx

```diff
@@ -44,3 +46,27 @@ export default class ErrorBoundary extends Component<PropsWithChildren<ErrorBoun
     return this.props.children
   }
 }
+
+/**
+ * This is a wrapper around the ErrorBoundary class component above.
+ * The class component handles regular Errors "thrown" during render.
+ *
+ * For other errors, which rely on reading from state, we need this functional component.
+ */
+export default function ErrorBoundary({ children, onError }: PropsWithChildren<ErrorBoundaryProps>) {
+  const connectionError = useAtomValue(connectionErrorAtom)
+  if (connectionError === ConnectionErrorType.MetamaskDisconnected) {
```

Comment:
I think this can be more generic so that it is more maintainable down the line - something like setting `error`/`header`/`action`/`onClick` based on the type of the error (using `instanceof`, see comments in `src/state/error.ts` about subclassing) instead of retyping the JSX for every case.